Converting the nginx.conf panics

Hi @mholt, thank you for Caddy and especially for writing this adapter.

Trying to run the adapter with the latest beta.20 panics now with following error:

### Error logs
```
./caddy adapt --config nginx.conf --adapter nginx
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/caddyserver/caddy/v2/caddyconfig.JSONModuleObject(0x13fabc0, 0x0, 0x15a1eb5, 0x7, 0x15a8319, 0x8, 0xc0007ce228, 0x1, 0x1357120, 0x1)
	github.com/caddyserver/caddy/v2@v2.0.0-beta.20/caddyconfig/configadapters.go:83 +0x441
github.com/caddyserver/nginx-adapter.(*setupState).locationContext(0xc0007cf968, 0xc0006f3110, 0xc0006146e0, 0x1, 0x1, 0xc000193a20, 0x1, 0x2, 0x27, 0x30, ...)
	github.com/caddyserver/nginx-adapter@v0.0.0-20200328151320-0464b93dab75/location.go:125 +0x3e43
github.com/caddyserver/nginx-adapter.(*setupState).serverContext(0xc0007cf968, 0xc0000e6b40, 0x5, 0x8, 0xc0007cf520, 0x1, 0x1, 0x1, 0x2)
	github.com/caddyserver/nginx-adapter@v0.0.0-20200328151320-0464b93dab75/server.go:110 +0x1efd
github.com/caddyserver/nginx-adapter.(*setupState).httpContext(0xc0007cf968, 0xc00023cd80, 0xb, 0x10, 0xc0007cf730, 0x1, 0x1, 0x1, 0x127da31)
	github.com/caddyserver/nginx-adapter@v0.0.0-20200328151320-0464b93dab75/nginxadapter.go:113 +0x4d4
github.com/caddyserver/nginx-adapter.(*setupState).mainContext(0xc0007cf968, 0xc0000e6480, 0x7, 0x8, 0x0, 0x0, 0xc00005e800, 0xc0007cf908, 0x43daec)
	github.com/caddyserver/nginx-adapter@v0.0.0-20200328151320-0464b93dab75/nginxadapter.go:87 +0x4cb
github.com/caddyserver/nginx-adapter.Adapter.Adapt(0xc0003c8a80, 0x774, 0x974, 0xc0006f29c0, 0xc00062e900, 0x1500000001516640, 0xc0007cfbc8, 0x159c3be7505eb59e, 0x10, 0x1356f20, ...)
	github.com/caddyserver/nginx-adapter@v0.0.0-20200328151320-0464b93dab75/nginxadapter.go:51 +0x1f6
github.com/caddyserver/caddy/v2/cmd.cmdAdaptConfig(0xc00018c9c0, 0xc0000d0140, 0x4, 0x4)
	github.com/caddyserver/caddy/v2@v2.0.0-beta.20/cmd/commandfuncs.go:464 +0x3ed
github.com/caddyserver/caddy/v2/cmd.Main()
	github.com/caddyserver/caddy/v2@v2.0.0-beta.20/cmd/main.go:84 +0x23e
main.main()
	caddy/main.go:12 +0x20
```

### Steps to reproduce:

```bash
go get github.com/caddyserver/builder/cmd/xcaddy

xcaddy build v2.0.0-beta.20 --with github.com/caddyserver/nginx-adapter@0464b93dab75db7fc111ad558d29615832b84f7f

curl https://raw.githubusercontent.com/caddyserver/nginx-adapter/master/testdata/example1.conf -o nginx.conf

./caddy adapt --config nginx.conf --adapter nginx
```

Edit: Updated the logs

--- a/processors.go
+++ b/processors.go
@@ -261,8 +261,17 @@ func processFastCGIPass(dirs []Directive) (*caddyhttp.Subroute, []caddyconfig.Wa
 		TransportRaw: caddyconfig.JSONModuleObject(fcgiTransport, "protocol", "fastcgi", nil),
 	}
 
+	/***
+	* If upstream doesn't have scheme explicitly specified as scheme://host, then
+	* assume it's tcp by prefixing it with tcp://. Later we check if the hostname is `unix:`.
+	* If so, then the arguemnt of fastcgi_pass was of the form `unix:/some/path`.
+	**/
 	passDirective, _ := getDirective(dirs, "fastcgi_pass")
-	upstream, err := url.Parse(passDirective.Param(1))
+	passArgs := passDirective.Param(1)
+	if !strings.Contains(passArgs, "://") {
+		passArgs = "tcp://" + passArgs
+	}
+	upstream, err := url.Parse(passArgs)
 	if err != nil {
 		warns = append(warns, caddyconfig.Warning{
 			File:      passDirective.File,
@@ -272,9 +281,12 @@ func processFastCGIPass(dirs []Directive) (*caddyhttp.Subroute, []caddyconfig.Wa
 		})
 		return nil, warns
 	}
-	network := "tcp"
-	host := upstream.Hostname()
-	if upstream.Hostname() == unixPrefix {
+
+	var network, host string = upstream.Scheme, upstream.Hostname()
+
+	// the argument of fastcgi_pass could have been of either of these forms:
+	// http://unix:/tmp/backend.socket:/uri/ , unix:///some/path
+	if upstream.Hostname() == unixPrefix || upstream.Scheme == "unix" {
 		network = "unix"
 		host = (strings.Split(upstream.Path, ":"))[0]
 	}

